### ISSUE TITLE
LinearLayout should optionally fill all available space in the main direction

### ISSUE THREAD

**Issue by keithkml** [mabe02/lanterna#484]
When a `Panel`'s child components are smaller than the available space (on the primary axis), and this option is enabled, the layout would automatically expand components evenly. This could work similarly to how it [shrinks them](https://github.com/mabe02/lanterna/blob/master/src/main/java/com/googlecode/lanterna/gui2/LinearLayout.java#L261). This might work best as a property on the layout data, rather than the layout manager itself.

**Comment by mabe02** [comment]
Agreed; I've done some experimental work on master for this, see the commit above

**Comment by keithkml** [comment]
Wow! Cool! This seems perfect.

**Comment by mabe02** [comment]
Alright, I can do another release shortly
